src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -1036,17 +1104,21 @@ private void ProcessStartOfFrameMarker(BufferedReadStream stream, int remaining,
                 JpegThrowHelper.ThrowInvalidImageContentException("Multiple SOF markers. Only single frame jpegs supported.");
             }
 
-            // Read initial marker definitions
+            // Read initial marker definitions.
             const int length = 6;
-            stream.Read(this.temp, 0, length);
+            int bytesRead = stream.Read(this.temp, 0, length);
```

Comment:
To double check: stream (in general) isn't guaranteed to read exactly the requested length, it may read up to length bytes. So this can be a latent bug, and usually there's a `while (bytesRead < length)`-loop.

I see the check below, and assume that BufferedReadStream's implemented takes this into mind (via CopyTo over there?).

So this is more or less a question to avoid a latent, nasty to find bug.

Reply:
Yeah `BufferedReadStream` uses a while loop to read internally.